allennlp/commands/fine_tune.py

```diff
@@ -166,6 +166,12 @@ def fine_tune_model(model: Model,
     test_data = all_datasets.get('test')
 
     trainer_params = params.pop("trainer")
+    nograd_regex_list = trainer_params.pop("no_grad", ())
+    if nograd_regex_list:
+        nograd_regex = "(" + ")|(".join(nograd_regex_list) + ")"
```

Comment:
this feels error-prone to me, I'd rather just 

```
no_grad_regexes = trainer_params.pop("no_grad", ())
for name, parameter in model.named_parameters():
    if any(re.search(regex, name) for regex in no_grad_regexes):
        parameter.requires_grad_(False)
```

it's cleaner and I can't imagine the time difference being noticeable

Reply:
Sure, will change that.